classes/class4/README.md

```diff
@@ -1 +1,484 @@
+You can include JavaScript directly in an HTML document using `<script>` tags,
+but writing any significant code within an HTML document would be silly.
+Instead we will reference a separate `.js` file in our HTML, similar to how we include CSS.
+
+Create a new file `main.js` in the `public/javascripts` directory, and paste in the following:
+```javascript
+console.log("hello world");
+```
+Then add this `<script>` tag to your `main.handlebars` layout file between the closing `</body>` and closing `</html>` tags like so:
+```html
+</body>
+<script type="text/javascript" src="javascripts/main.js"></script>
+</html>
+```
+When HTML is parsed, it is rendered top-down.
+When a script tag is hit, its contents are run immediately.
+This means that if your JavaScript references part of the DOM
+it may not exist yet and your script will not run as expected.
+There are some ways to force your script to wait until the page is ready before running,
+but a simple way around this is to just add your `<script>` tag below the HTML body.
+
+Now, when you refresh the page, open up the developer console and you should see:
+```javascript
+hello world
+```
+Being able to print to the console and do math and such is nice,
+but really we want to interact with the DOM.
+To do that we need to dig into the `document` object.
+In the dev console run:
+```javascript
+> document.body.children[1].innerHTML
+"GET request"
+```
+That returns the contents of the 2nd child of the `<body>` tag, which in this case is
+`<h2>GET request</h2>`.
+
+We can assigned the `innerHTML` parameter to a new value:
+```javascript
+document.body.children[1].innerHTML = "Nothing at all..."
+```
+and the webpage will change accordingly.
+
+But that's pretty much the extent to which knowing the structure of the
+`document` object is particularly useful.
+The API of the DOM is a bit dense and not exactly user friendly.
+Feel free to read more about it [here](https://developer.mozilla.org/en-US/docs/Web/API/Document_Object_Model),
+but for now we are going to let other people do the hard stuff for us and make
+editing the DOM a lot easier on ourselves.
+This is where [jQuery](http://jquery.com/) comes in handy.
+jQuery is a fairly ubiquitous library that makes DOM manipulation, event handling, animation,
+and AJAX much easier.
+
+We will include jQuery in the client by adding the line:
+```html
+<script src="https://ajax.googleapis.com/ajax/libs/jquery/2.1.3/jquery.min.js"></script>
+```
+To the `<head>` section of `main.handlebars`.
+In the case of JavaScript dependencies, we include them early because they
+are not dependent on the contents of the `<body>` and it helps to keep them organized and easily findable.
+
+If you refresh the page, nothing will appear different, but you now have a powerful tool at your disposal, the `$`.
+You can access the jQuery object in two ways `jQuery()` or `$()`, but everyone uses the `$`.
```

Comment:
The history here is that people once-upon-a-time hated typing `document` so often and would reassign it to `$`. When jQuery came around they assumed you wouldn't need `document` anymore and so took the convention for themselves.

I read a lot of stuff on the internet.